skellytracker/trackers/openpose_tracker/openpose_tracker.py

```diff
@@ -0,0 +1,129 @@
+        :param number_people_max: Maximum number of people to detect.
+        """
+        # super().__init__(recorder=recorder)
+        self.openpose_exe_path = Path(openpose_exe_path)
+        self.output_json_path = Path(output_json_path)
+        self.net_resolution = net_resolution # TODO: this and num_people should be parameters for process_video, since we could use this one tracker to process videos with different parameters
+        self.number_people_max = number_people_max
+
+        super().__init__(
+            tracked_object_names=[],
+            recorder=OpenPoseRecorder(json_directory_path=output_json_path),
+        )
+
+        os.chdir(self.openpose_exe_path) # TODO: this can mess with things downstream if we're not sure to change the path back, we should just be explicit about the path throughout
+
+    def process_video(
+        self,
+        input_video_filepath: Union[str, Path],
+        output_video_filepath: Union[str, Path],
+        save_data_bool: bool = False,
+        use_tqdm: bool = True,
+        **kwargs,
+    ):
+        """
+        Run the OpenPose demo on a video file to generate JSON outputs
+        in a unique directory for each video.
+        """
+
+        # TODO: Add some check that the openpose executable exists (and maybe do a cross platform search like we do for blender in fmc?)
+
+        # Extract video name without extension to use as a unique folder name
+        video_name = Path(input_video_filepath).stem
+        unique_json_output_path = self.output_json_path / video_name
+        unique_json_output_path.mkdir(
+            parents=True, exist_ok=True
+        )  # Create the directory if it doesn't exist
+
+        # video_save_path = Path(output_video_filepath) / f"{video_name}_openpose.avi"
+
+        # Update the subprocess command to use the unique output directory
+        # TODO: subprocess call should probably be in a try/except with some form of error handling
+        subprocess.run(
+            [
+                "bin\OpenPoseDemo.exe",  # TODO: this is throwing the error `S607 Starting a process with a partial executable path`, needs to be something like usr/bin/OpenPoseDemo.exe, but I can't verify where it is (also does it need two backslashes to escape properly?)
+                "--video",
+                str(input_video_filepath),
+                "--write_json",
+                str(unique_json_output_path),
+                "--net_resolution",
+                self.net_resolution,
+                "--hand",
+                "--face",
+                "--number_people_max",
+                str(self.number_people_max),
+                "--write_video",
+                str(output_video_filepath),
+                "--output_resolution",
+                "-1x-1",
+            ],
+            shell=True,  # TODO: This is generally frowned upon if there's not a reason for it, see `S602 subprocess call with shell=True identified, security issue.`
```

Comment:
is there a reason for it here?

Reply:
Lol this is my fault for not doing so - but a lot of the comments in this specific file are ones that Jon made *years* ago in the original pre-alpha mediapipe openpose code that I copied over into our current version, and I should have deleted them. However, in this case, I did just check to see if `shell=False` works properly, and it seems like it does - so we can safely change this I'd say.